### PR TITLE
Throwing error when the configs fail JSON serialization

### PR DESCRIPTION
**Pull request by kevin36524**
Fix Issue https://github.com/huggingface/swift-transformers/issues/112

Some HF models like mistral needs `HUGGING_FACE_HUB_TOKEN` to be set for the tokenizer and model config to be downloaded. 

Currently we are getting this generic error if the HUGGING_FACE_HUB_TOKEN is not set.

```
(venv) kevin@Mac swift-transformers % swift run transformers "Best recommendations for a place to visit in Paris in August 2024:" --max-length 128 Examples/Mistral7B/StatefulMistral7BInstructInt4.mlpackage

Building for debugging...
[1/1] Write swift-version-39B54973F684ADAB.txt
Build of product 'transformers' complete! (0.17s)
Compiling model Examples/Mistral7B/StatefulMistral7BInstructInt4.mlpackage -- file:///Users/arforgeqa/Projects/swift-transformers/
Loading model file:///var/folders/4x/2n2whrg17s7g6s9xcr67tmt80000gn/T/StatefulMistral7BInstructInt4_A3224709-606B-4B0E-B497-34BD3B7C7FA1.mlmodelc
Generating
Error: The data couldn’t be read because it isn’t in the correct format.
```

This PR add some nice error logs to give more hints to the user.

New Output

```
kevin@Mac swift-transformers % swift run transformers "Write me an email wishing merry christmas" --max-length 20 ../kmodels/StatefulMistral7BInstructInt4.mlpackage
Building for debugging...
[11/11] Applying transformers
Build of product 'transformers' complete! (1.63s)
Compiling model ../kmodels/StatefulMistral7BInstructInt4.mlpackage -- file:///Users/kevin/projects/swift-transformers/
Loading model file:///var/folders/wl/tr847nmj2sz6b_w6qyyxmbr80000gn/T/StatefulMistral7BInstructInt4_8F1E98A9-413D-443D-B9FD-6F7A6B2AAA19.mlmodelc
Generating
Error: JSON Serialization failed for file:///Users/kevin/Documents/huggingface/models/mistralai/Mistral-7B-Instruct-v0.3/config.json. Please verify that you have set the HUGGING_FACE_HUB_TOKEN environment variable.
```

### CODE REVIEW

**Review by pcuenca — APPROVED** [review]
Thank you! Conceptually looks good to me, but waiting a bit to merge:

- I think HUGGING_FACE_HUB_TOKEN will not be supported until #111 is merged.
- I'm slightly concerned about potential clients of `HubApi` receiving a different error now, but I suppose it should be ok.
- I'd like to run the tests offline. We are planning to restore CI next week, hopefully, but meanwhile I'll do it locally.

File: Sources/Hub/HubApi.swift [hunk 1]
```diff
@@ -106,6 +106,13 @@ public extension HubApi {
     /// `fileURL` is a complete local file path for the given model
     func configuration(fileURL: URL) throws -> Config {
         let data = try Data(contentsOf: fileURL)
+        do {
+            let parsed = try JSONSerialization.jsonObject(with: data, options: [])
+            guard let dictionary = parsed as? [String: Any] else { throw Hub.HubClientError.parse }
```
Comment (pcuenca) [inline Sources/Hub/HubApi.swift:111]:
```suggestion
            guard let dictionary = parsed as? [NSString: Any] else { throw Hub.HubClientError.parse }
```

My bad, made a mistake in the merge

File: Sources/Hub/HubApi.swift [hunk 2]
```diff
@@ -106,6 +106,13 @@ public extension HubApi {
...
+            return Config(dictionary)
+        } catch {
+            throw Hub.HubClientError.jsonSerialization(fileURL: fileURL, message: "JSON Serialization failed for \(fileURL). Please verify that you have set the HUGGING_FACE_HUB_TOKEN environment variable.")
+        }
         let parsed = try JSONSerialization.jsonObject(with: data, options: [])
         guard let dictionary = parsed as? [NSString: Any] else { throw Hub.HubClientError.parse }
         return Config(dictionary)
```
Comment (pcuenca) [inline Sources/Hub/HubApi.swift:118]:
```suggestion
```

**Comment by pcuenca** [comment]
Tests pass locally. Thanks again @kevin36524!

**Comment by kevin36524** [comment]
That's great thanks @pcuenca

**Pull request merged**